flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -263,22 +265,54 @@ public static int getDefaultMaxParallelism(JobVertex vertex) {
                 normalizeParallelism(vertex.getParallelism()));
     }
 
+    // TODO: move to Adaptive Batch Scheduler.
+    /**
+     * Compute the {@link VertexParallelismStore} for all given vertices in a dynamic graph, which
+     * will set defaults and ensure that the returned store contains valid parallelisms, with the
+     * configured default max parallelism.
+     *
+     * @param vertices the vertices to compute parallelism for
+     * @param defaultMaxParallelism the global default max parallelism
+     * @return the computed parallelism store
+     */
+    @VisibleForTesting
+    public static VertexParallelismStore computeVertexParallelismStoreForDynamicGraph(
```

Comment:
I would suggest to add this method to `ExecutionJobVertexTest` for now and later add it to `AdaptiveBatchScheduler`.